tests/integration/targets/ip_assign/tasks/main.yml

```diff
@@ -0,0 +1,52 @@
+- name: ip_assign
+  block:
+    - set_fact:
+        r: "{{ 1000000000 | random }}"
+    - name: Create a Linode Instance
+      linode.cloud.instance:
+        label: 'ansible-test-{{ r }}'
+        region: us-southeast
+        type: g6-standard-1
+        image: linode/alpine3.17
+        state: present
+      register: create_instance
+
+    - name: Create another Linode Instance
+      linode.cloud.instance:
+        label: 'ansible-test-{{ r }}-2'
+        region: us-southeast
+        type: g6-standard-1
+        image: linode/alpine3.17
+        state: present
+      register: create_instance_2
+
+    - name: Swap Both IPs
+      linode.cloud.ip_assign:
+        region: us-southeast
+        assignments:
+          - address: '{{ create_instance.networking.ipv4.public[0].address }}'
+            linode_id: '{{ create_instance_2.instance.id }}'
+          - address: '{{ create_instance_2.networking.ipv4.public[0].address }}'
+            linode_id: '{{ create_instance.instance.id }}'
+
+    - name: Get updated Instance1
+      linode.cloud.instance_info:
+        id: '{{ create_instance.instance.id }}'
+      register: updated_instance
+
+    - name: Get updated Instance2
+      linode.cloud.instance_info:
+        id: '{{create_instance_2.instance.id }}'
+      register: updated_instance_2
+
+    - name: Assert changes
+      assert:
+        that:
+          - updated_instance.networking.ipv4.public[0].address == create_instance_2.networking.ipv4.public[0].address
+          - updated_instance_2.networking.ipv4.public[0].address == create_instance.networking.ipv4.public[0].address
+
```

Comment:
We probably can add a block like this to destroy the testing instances in the end:
```
  always:
    - ignore_errors: true
      block:
        - linode.cloud.instance:
            label: '{{ instance_create.instance.label }}'
            state: absent
```